src/platform/nrfconnect/wifi/NrfWiFiDriver.cpp

```diff
@@ -276,20 +279,13 @@ void NrfWiFiDriver::ScanNetworks(ByteSpan ssid, WiFiDriver::ScanCallback * callb
     }
 }
 
-CHIP_ERROR NrfWiFiDriver::GetSupportedWiFiBands(Span<WiFiBand> & bands)
+uint32_t NrfWiFiDriver::GetSupportedWiFiBands()
 {
-    static constexpr WiFiBand kBands[] = {
-        WiFiBand::k2g4,
+    uint32_t bands = static_cast<uint32_t>(1UL << chip::to_underlying(WiFiBandEnum::k2g4));
```

Comment:
here is where a BitMask would be nice:

```cpp
BitMask<WiFiBandEnum, uint32_t> bands(WiFiBandEnum::k2g4);

#ifndef CONFIG_BOARD_NRF7001
    bands.Set(WiFiBandEnum::k5g);
#endif

return bands;
```

Reply:
we may not even need the 2nd argument since to_underlying will auto-adjust

Reply:
It's not a linear enum